README.md

```diff
@@ -187,24 +187,30 @@ $ curl -X DELETE "localhost:8080/delete_all"
 
 ### Query Endpoints
 
-These endpoints match those used by the Grafana Simple JSON datasource.
+These endpoints match those used by the [Grafana Simple JSON datasource](https://grafana.com/grafana/plugins/grafana-simple-json-datasource/).
 
 #### GET /search
 
-Responds with a JSON array containing the selectable query elements.
+Responds with a JSON array containing selectable values of an event field (e.g. `jdk.ObjectAllocationSample.objectClass`) specified in the JSON body's `target` field. Used to define available selections for [dashboard variables](https://grafana.com/docs/grafana/v7.5/variables/).
+
+If `target` is set to `*`, responds with all selectable event fields.
+
 
 CURL Example
 ```bash
-$ curl "localhost:8080/search"
+$ curl -X POST --data '{ "target": "jdk.ObjectAllocationSample.objectClass" }' "localhost:8080/search"
 ```
 
+
 #### POST /query
 
-Responds with a JSON array containing elements for a query. The query body format matches that of the Grafana Simple JSON datasource.
+Responds with a JSON array containing data points for a query. The query body format matches that of the Grafana Simple JSON datasource.
+
+The `target` field can have parameters to filter matching data points If there is no paramater, no matching is performed. If a parameter is specified with "*", matching is done for all possible value of that paramter.
```

Comment:
s/paramater/parameter/
s/paramter/parameter/

This uses query parameter syntax for the `?` delimiter - does it also support `&` for conjunction of queries?

Reply:
Yes, it does support `&` for conjunction of queries. For example:

`{ "target": "jdk.ObjectAllocationSample.weight?objectClass=java.util.ArrayList&objectClass=java.lang.String" }`

`{ "target": "jdk.ObjectAllocationSample.weight?weight=java.util.ArrayList&startTime = 18:44:30.638" }`

Each JSON object in the returned array has a `meta` field (JSON object) that has a field called `paramField` to specify the name of the parameter.